src/cargo/ops/cargo_compile/mod.rs

```diff
@@ -476,6 +480,49 @@ pub fn create_bcx<'a, 'gctx>(
             .extend(args);
     }
 
+    {
+        let check_cfg = target_data.check_cfg();
+
+        if check_cfg.exhaustive {
+            let target_cfgs = gctx.target_cfgs()?;
+            for (key, _fields) in target_cfgs {
+                if !key.starts_with("cfg(") || !key.ends_with(')') {
+                    continue;
+                }
+
+                let cfg_expr = &key[4..key.len() - 1];
+                let Ok(cfg_expr) = CfgExpr::from_str(cfg_expr) else {
+                    continue;
+                };
```

Comment:
Thats a
```rust
let Some(key) = key.strip_prefix("cfg(") else { continue ; };
let Some(key) = key.strip_suffix(")") else { continue ; };
```

Even better is to not re-implement this parser and instead use `Platform::from_str`